### DISCUSSION TITLE
转换工具 PAGConvertor 转换 MP4 和图片帧失败，是有什么依赖吗？

### DISCUSSION THREAD

**Discussion by CxmyDev** [Tencent/libpag#1986] [Q&A]
【版本信息】
【PAG 转换工具下载】，官网直接下载的。

【平台信息】
Mac OS。

【预期的表现】
直接下载的工具，里面的 Demo，尝试转换 input/pic 的图片帧 和 MP4 文件转 PAG ，均报错。
不确定是否还需要安装什么依赖。

图片帧转 PAG 报错。
admin@admindeMacBook-Pro-5 mac2 % ./convert.sh
begin
cd input
pic
input file :pic/ frame count: 25.000000
Could not open input file 'pic/meiwei_000.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_000.png
Could not open input file 'pic/meiwei_001.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_001.png
Could not open input file 'pic/meiwei_002.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_002.png
Could not open input file 'pic/meiwei_003.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_003.png
Could not open input file 'pic/meiwei_004.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_004.png
Could not open input file 'pic/meiwei_005.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_005.png
Could not open input file 'pic/meiwei_006.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_006.png
Could not open input file 'pic/meiwei_007.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_007.png
Could not open input file 'pic/meiwei_008.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_008.png
Could not open input file 'pic/meiwei_009.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_009.png
Could not open input file 'pic/meiwei_010.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_010.png
Could not open input file 'pic/meiwei_011.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_011.png
Could not open input file 'pic/meiwei_012.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_012.png
Could not open input file 'pic/meiwei_013.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_013.png
Could not open input file 'pic/meiwei_014.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_014.png
Could not open input file 'pic/meiwei_015.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_015.png
Could not open input file 'pic/meiwei_016.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_016.png
Could not open input file 'pic/meiwei_017.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_017.png
Could not open input file 'pic/meiwei_018.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_018.png
Could not open input file 'pic/meiwei_019.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_019.png
Could not open input file 'pic/meiwei_020.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_020.png
Could not open input file 'pic/meiwei_021.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_021.png
Could not open input file 'pic/meiwei_022.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_022.png
Could not open input file 'pic/meiwei_023.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_023.png
Could not open input file 'pic/meiwei_024.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_024.png
Could not open input file 'pic/meiwei_025.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_025.png
Could not open input file 'pic/meiwei_026.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_026.png
Could not open input file 'pic/meiwei_027.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_027.png
Could not open input file 'pic/meiwei_028.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_028.png
Could not open input file 'pic/meiwei_029.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_029.png
Could not open input file 'pic/meiwei_030.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_030.png
Could not open input file 'pic/meiwei_031.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_031.png
Could not open input file 'pic/meiwei_032.png' (error 'Invalid argument')
can't create decoder :pic/meiwei_032.png
Could not open input file 'pic/meiwei_033.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_033.png
Could not open input file 'pic/meiwei_034.png' (error 'Invalid argument')
can't create decoder :pic/meiwei_034.png
Could not open input file 'pic/meiwei_035.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_035.png
Could not open input file 'pic/meiwei_036.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_036.png
Could not open input file 'pic/meiwei_037.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_037.png
Could not open input file 'pic/meiwei_038.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_038.png
Could not open input file 'pic/meiwei_039.png' (error 'Invalid argument')
can't create decoder :pic/meiwei_039.png
Could not open input file 'pic/meiwei_040.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_040.png
Could not open input file 'pic/meiwei_041.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_041.png
Could not open input file 'pic/meiwei_042.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_042.png
Could not open input file 'pic/meiwei_043.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_043.png
Could not open input file 'pic/meiwei_044.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_044.png
Could not open input file 'pic/meiwei_045.png' (error 'Invalid argument')
can't create decoder :pic/meiwei_045.png
Could not open input file 'pic/meiwei_046.png' (error 'Invalid data found when processing input')
can't create decoder :pic/meiwei_046.png

Export failed. PAGFile generation failed
error : make video sequence failed
end

尝试转换 input/test.mp4 的视频，报错。
admin@admindeMacBook-Pro-5 mac2 % ./convert.sh
begin
cd input
test.mp4
./convert.sh: line 10:  1388 Segmentation fault: 11  ../PAGConvertor ${file}
end

【Demo及附件】
直接下载官方 Demo 就可以复现。
https://pag.art/docs/pag-convert-download.html

**Comment by domchen** [comment]
我刚刚测试了一下是正常的没有报错，设备是M1。你的电脑硬件和系统操作版本是多少？另外检查一下系统设置看看是不是权限问题。macOS会禁止第三方未知程序执行。

**Reply by xiaoyaogaobin** [reply]
<img width="321" alt="image" src="https://github.com/Tencent/libpag/assets/44795293/9b35cea4-ebd0-4af6-8cda-48fb724872b7">
我的系统版本是这个，执行官方demo，也是一样的错误

**Comment by achenyingpeng** [comment]
这边遇到了相同的问题，M1的设备，转换mp4没问题但是转换png序列帧出现如上的报错
Could not open input file 'fire/fire_00236.png' (error 'Invalid data found when processing input')
can't create decoder :fire/fire_00236.png
Could not open input file 'fire/fire_00237.png' (error 'Invalid data found when processing input')
can't create decoder :fire/fire_00237.png
Could not open input file 'fire/fire_00238.png' (error 'Invalid data found when processing input')
can't create decoder :fire/fire_00238.png
Could not open input file 'fire/fire_00239.png' (error 'Invalid data found when processing input')
can't create decoder :fire/fire_00239.png
Could not open input file 'fire/fire_00240.png' (error 'Invalid data found when processing input')
can't create decoder :fire/fire_00240.png

有解决办法吗？

**Reply by achenyingpeng** [reply]
https://superuser.com/questions/915302/ffmpeg-png-jpg-invalid-data-found-when-processing-input?rq=1

**Reply by achenyingpeng** [reply]
附带一个ffmpeg的类似的报错，看下有没有参考作用🙏

**Reply by ndNovaDev** [reply]
intel机器，和你一样的问题

**Reply by domchen** [reply]
根据前面链接的提示，有可能是系统缺少 zlib 库，所以解码不了 PNG 图片，可以命令行执行 `brew install zlib`试试看。

**Reply by ndNovaDev** [reply]
zlib 1.3.1 is already installed and up-to-date

问题还在

**Reply by JJS-CN** [reply]
m1机器, 权限全给, zlib安装也一样的错误

**Reply by IsidoreOC5869265asey** [reply]
M3 的序列帧转换成 pag 也遇到了同样的问题

**Reply by alstone** [reply]
。。。

**Reply by 51lucky** [reply]
同样的问题

**Comment by CodeJhF** [comment]
之前请你补充的复现信息（demo / 版本号 / 机型 / 日志）一直没有收到，缺少这些信息无法继续定位，先关闭本条。
如果问题还在，欢迎补齐上述信息后回复或重开，我们会优先跟进。（stale-close）
